tests/simulators/carla/test_carla.py

```diff
@@ -8,6 +9,412 @@
 )
 
 
+def preprocess_old_blueprint_names(original):
+    d = {}
+    for key, value in original.items():
+        for sub_value in value:
+            d[sub_value] = key
+    return d
+
+
+def model_blueprint(simulator, mapPath, town, modelType, modelName, newModelNames=None):
+    code = f"""
+            param map = r'{mapPath}'
+            param carla_map = '{town}'
+            param time_step = 1.0/10
+
+            model scenic.simulators.carla.model
+            ego = new {modelType} with blueprint '{modelName}'
+            terminate after 1 steps
+        """
+    scenario = compileScenic(code, mode2D=True)
+    scene, _ = scenario.generate(maxIterations=10000)
```

Comment:
```suggestion
    scene = sampleScene(scenario)
```
The `sampleScene` helper does some additional checks, and I don't think there's any reason to allow more than 1 iteration (these scenarios should not cause any rejections).